src/main/java/com/scalefocus/blogapp/interceptor/JwtInterceptor.java

```diff
@@ -0,0 +1,62 @@
+package com.scalefocus.blogapp.interceptor;
+
+import com.scalefocus.blogapp.handler.SessionHandler;
+import com.scalefocus.blogapp.service.JwtServiceImpl;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import java.text.ParseException;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class JwtInterceptor implements HandlerInterceptor {
+
+    private final JwtServiceImpl jwtService;
+
+    private final SessionHandler sessionHandler;
+
+    private static final String TOKEN_PREFIX = "Bearer ";
+
+    @Override
+    public boolean preHandle(HttpServletRequest request,
+                             HttpServletResponse response,
+                             Object handler) throws ParseException {
+
+        final String authorization = request.getHeader("Authorization");
+
+        if (authorization.isBlank() || Boolean.FALSE.equals(authorization.startsWith(TOKEN_PREFIX))) {
+
+            return unauthorized(response);
+
+        }
+
+        final String jwt = authorization.substring(TOKEN_PREFIX.length());
+        final String username = jwtService.extractUsername(jwt);
+        final String userId = jwtService.extractUserId(jwt);
+
+        if (Boolean.FALSE.equals(jwtService.validateToken(jwt, username))) {
```

Comment:
do you compare results from same methods?

Reply:
No, there is no comparison, **validateToken** is a boolean method if JWT is not valid according to the username it returns false.